Microchip/Winc15x0/GHIElectronics.TinyCLR.Drivers.Microchip.Winc15x0/Winc15x0Interface.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Runtime.CompilerServices;
+using GHIElectronics.TinyCLR.Devices.Network;
+
+namespace GHIElectronics.TinyCLR.Drivers.Microchip.Winc15x0 {
+    public class Winc15x0Interface {
+        public NetworkController NetworkController { get; }
+
+        private bool initialized = false;
+
+        public Winc15x0Interface() => this.NetworkController = NetworkController.FromName("GHIElectronics.TinyCLR.NativeApis.ATWINC15xx.NetworkController");
+
+        ~Winc15x0Interface() => this.Dispose();
+
+        public void Dispose() {
+            this.NetworkController.Dispose();
+            GC.SuppressFinalize(this);
+        }
+
+        public string GetFirmwareVersion() {
+            if (!this.initialized) {
+                this.NativeTurnOn();
+
+                this.initialized = true;
+            }
+
+            this.NativeReadFirmwareVersion(out var ver1, out var ver2);
+
+            var major = (ver1 >> 16) & 0xFF;
+            var minor = (ver1 >> 8) & 0xFF;
+            var patch = (ver1 >> 0) & 0xFF;
+
+            return $"{major}.{minor}.{patch}.{ver2}";
+        }
+
+        public bool FirmwareUpdatebyOta(string url, int timeout) {
+            if (!this.initialized) {
```

Comment:
Abstract this functionality out to a private TurnOn method that does exactly this